src/pages/index.tsx

```diff
@@ -36,6 +36,9 @@ export default function Page({ providers }: Props) {
       request<LoadUserInfoParams, LoadUserInfoResults>('/loadUserInfo', {}).then(res => {
         if (res) {
           const { user, classes } = res
+          if (!user.studentID) {
+            push('/registration/studentID')
```

Comment:
I think this should go after the dispatch.
Or better, we can give users more control on when they will enter their id by making a button to move tot he registration page.